### PR TITLE
I ADDED WINTER COATS BUT NEED TO ADD HOODS

### PR DESCRIPTION
**Pull request by Gef21**
HOODS SOON(ISH)

PROOF THAT IT WORKS AND IS TESTED https://webmshare.com/vEKAW

THIS TOOK ME MORE THAN TEN HOURS AND IN THE END ALL I DID WAS PORT SOME SPRITES HAHAHAHAHA KILL ME PLEASE

### CODE REVIEW

File: code/modules/clothing/suits/toggles.dm [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    /obj/item/weapon/paper,
+    /obj/item/device/flashlight,
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+/obj/item/clothing/suit/wintercoatcargo
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Make this inherit from winter coat and remove like 90% of the lines here

File: code/modules/clothing/suits/toggles.dm [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    /obj/item/weapon/paper,
+    /obj/item/device/flashlight,
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+/obj/item/clothing/suit/wintercoatscience
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+    /obj/item/weapon/paper,
+    /obj/item/device/flashlight,
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+/obj/item/clothing/suit/wintercoatengineering
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

**Comment by Gef21** [comment]
It passed all the checks which means you have to merge it without reading it first

File: code/modules/clothing/suits/toggles.dm [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    /obj/item/weapon/paper,
+    /obj/item/device/flashlight,
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+/obj/item/clothing/suit/wintercoatatmos
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+    /obj/item/device/flashlight,
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+
+/obj/item/clothing/suit/wintercoatmining
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

**Review by ghost — CHANGES_REQUESTED** [review]
TEN HOURS

File: code/modules/clothing/suits/toggles.dm (hunk 1 above)
Comment (Gef21) [reply code/modules/clothing/suits/toggles.dm]: I don't know what that means

File: code/modules/clothing/suits/toggles.dm (hunk 2 above)
Comment (Gef21) [reply code/modules/clothing/suits/toggles.dm]: I don't know what that means

File: code/modules/clothing/suits/toggles.dm (hunk 3 above)
Comment (Gef21) [reply code/modules/clothing/suits/toggles.dm]: I don't know what that means

File: code/modules/clothing/suits/toggles.dm (hunk 4 above)
Comment (Gef21) [reply code/modules/clothing/suits/toggles.dm]: I don't know what that means

File: code/modules/clothing/suits/toggles.dm (hunk 5 above)
Comment (Gef21) [reply code/modules/clothing/suits/toggles.dm]: I don't know what that means

File: code/modules/clothing/suits/toggles.dm (hunk 1 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm]: 😦

File: code/modules/clothing/suits/toggles.dm (hunk 2 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm]: 😦

File: code/modules/clothing/suits/toggles.dm (hunk 3 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm]: 😦

File: code/modules/clothing/suits/toggles.dm (hunk 4 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm]: 😦

File: code/modules/clothing/suits/toggles.dm (hunk 5 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm]: 😦

**Comment by Zennyization** [comment]
🎉 I support this PR 🎉 

 👍 LONG LIVE GEF 👍

**Comment by TheFriggenSawbones** [comment]
:tada: I support this PR :tada:

:+1: LONG LIVE GEF :+1:

File: code/modules/clothing/suits/toggles.dm [hunk 6]
```diff
@@ -0,0 +1,64 @@
+/obj/item/clothing/suit/wintercoat
+    name = "winter coat"
+    desc = "A warm, fluffy and incredibly comfortable jacket made from animal furs."
+    icon_state = "coatwinter"
+    var item_state_slots = list(slot_r_hand_str = "coatwinter", slot_l_hand_str = "coatwinter")
```
Comment (Intigracy) [inline code/modules/clothing/suits/toggles.dm:5]: what is this

**Comment by ghost** [comment]
One hour for every ten lines of code

File: code/modules/clothing/suits/toggles.dm (hunk 6 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm:5]: Copy pasted vore code I believe

**Comment by Gef21** [comment]
15~ days for one bug :^)

File: code/modules/clothing/suits/toggles.dm [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    /obj/item/device/flashlight,
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+/obj/item/clothing/suit/wintercoat/cargo
+    name = "cargo technicians winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Don't like this name, doesn't flow, and is grammatically incorrect without an apostrophe

File: code/modules/clothing/suits/toggles.dm [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatcargo"
+
+
+/obj/item/clothing/suit/wintercoat/science
+    name = "scientists winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same tbh

File: code/modules/clothing/suits/toggles.dm [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatscience"
+
+
+/obj/item/clothing/suit/wintercoat/medical
+    name = "doctors winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatmedical"
+
+
+/obj/item/clothing/suit/wintercoat/captain
+    name = "captains winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatcaptain"
+
+
+/obj/item/clothing/suit/wintercoat/engineering
+    name = "engineers winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatengineering"
+
+
+/obj/item/clothing/suit/wintercoat/hydro
+    name = "gardeners winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coathydro"
+
+
+/obj/item/clothing/suit/wintercoat/atmos
+    name = "atmos technicians winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatatmos"
+
+
+
+/obj/item/clothing/suit/wintercoat/security
+    name = "securitys winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 15]
```diff
@@ -0,0 +1,64 @@
...
+    icon_state = "coatsecurity"
+
+
+/obj/item/clothing/suit/wintercoat/mining
+    name = "miners winter coat"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm]: Same

File: code/modules/clothing/suits/toggles.dm (hunk 12 above)
Comment (The-Albinobigfoot) [reply code/modules/clothing/suits/toggles.dm]: > gardener

File: code/modules/clothing/suits/toggles.dm (hunk 13 above)
Comment (The-Albinobigfoot) [reply code/modules/clothing/suits/toggles.dm]: > atmos

File: code/modules/clothing/suits/toggles.dm (hunk 6 above)
Comment (ghost) [reply code/modules/clothing/suits/toggles.dm:5]: @Gef21 what he means is you should remove this and replace it with the actual item states and in hand states

File: code/modules/clothing/suits/toggles.dm [hunk 16]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/obj/item/clothing/suit/wintercoat/medical
+    name = "medical winter coat"
+    icon_state = "coatmedical"
+
+
+/obj/item/clothing/suit/wintercoat/captain
+    name = "captain winter coat"
+    icon_state = "coatcaptain"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:39]: You forgot the item state for this, did you even test it.?

File: code/modules/clothing/suits/toggles.dm [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/obj/item/clothing/suit/wintercoat/science
+    name = "science winter coat"
+    icon_state = "coatscience"
+
+
+/obj/item/clothing/suit/wintercoat/medical
+    name = "medical winter coat"
+    icon_state = "coatmedical"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:34]: Ditto

File: code/modules/clothing/suits/toggles.dm [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/obj/item/clothing/suit/wintercoat/cargo
+    name = "cargo winter coat"
+    icon_state = "coatcargo"
+
+
+/obj/item/clothing/suit/wintercoat/science
+    name = "science winter coat"
+    icon_state = "coatscience"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:29]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+    /obj/item/weapon/tank/emergency_oxygen,
+    /obj/item/weapon/storage/fancy/cigarettes,
+//    /obj/item/weapon/storage/box/matches,
+    /obj/item/weapon/reagent_containers/food/drinks/flask
+    )
+
+
+/obj/item/clothing/suit/wintercoat/cargo
+    name = "cargo winter coat"
+    icon_state = "coatcargo"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:24]: Did you forget the item states? Yeah you did

File: code/modules/clothing/suits/toggles.dm [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/obj/item/clothing/suit/wintercoat/captain
+    name = "captain winter coat"
+    icon_state = "coatcaptain"
+
+
+/obj/item/clothing/suit/wintercoat/engineering
+    name = "engineering winter coat"
+    icon_state = "coatengineering"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:44]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/obj/item/clothing/suit/wintercoat/security
+    name = "security winter coat"
+    icon_state = "coatsecurity"
+
+
+/obj/item/clothing/suit/wintercoat/mining
+    name = "mining winter coat"
+    icon_state = "coatmining"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:65]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+
+/obj/item/clothing/suit/wintercoat/atmos
+    name = "atmos winter coat"
+    icon_state = "coatatmos"
+
+
+
+/obj/item/clothing/suit/wintercoat/security
+    name = "security winter coat"
+    icon_state = "coatsecurity"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:60]: Same

File: code/modules/clothing/suits/toggles.dm [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+
+
+/obj/item/clothing/suit/wintercoat/hydro
+    name = "hydroponics winter coat"
+    icon_state = "coathydro"
+
+
+/obj/item/clothing/suit/wintercoat/atmos
+    name = "atmos winter coat"
+    icon_state = "coatatmos"
```
Comment (ghost) [inline code/modules/clothing/suits/toggles.dm:54]: Same

**Review by ghost — CHANGES_REQUESTED** [review]

File: code/modules/clothing/suits/toggles.dm [hunk 24]
```diff
@@ -19,46 +19,46 @@
...
 
 /obj/item/clothing/suit/wintercoat/hydro
-    name = "gardeners winter coat"
+    name = "hydroponics winter coat"
     icon_state = "coathydro"
 
 
 /obj/item/clothing/suit/wintercoat/atmos
-    name = "atmos technicians winter coat"
+    name = "atmos winter coat"
```
Comment (The-Albinobigfoot) [inline code/modules/clothing/suits/toggles.dm]: > atmos

File: code/modules/clothing/clothing.dm [hunk 25]
```diff
@@ -329,6 +329,38 @@ BLIND     // can't see anything
 	body_parts_covered = HEAD
 	slot_flags = SLOT_HEAD
 	species_restricted = list("exclude","Muton")
+		// this is for hoods
```
Comment (clusterfack) [inline code/modules/clothing/clothing.dm]: This is not how you comment an entire block of code, you start it with ```/*``` and end it with ```*/```

File: code/modules/clothing/suits/toggles.dm (hunk 16 above)
Comment (Gef21) [reply code/modules/clothing/suits/toggles.dm:39]: I think the item states are handled somewhere else

File: code/modules/clothing/clothing.dm (hunk 25 above)
Comment (Exxion) [reply code/modules/clothing/clothing.dm]: Fuck you Cluster
Block comments fuck up Git
This is a better method

**Comment by Gef21** [comment]
Including the item state makes no difference for some reason so I'm leaving it out

Proofs https://webmshare.com/078Kj

**Comment by Zennyization** [comment]
🎉 I support this PR 🎉

👍 LONG LIVE GEF 👍

**Comment by Zennyization** [comment]
🎉 I support this PR 🎉

👍 LONG LIVE GEF 👍

File: code/modules/clothing/suits/toggles.dm (hunk 16 above)
Comment (ComicIronic) [reply code/modules/clothing/suits/toggles.dm:39]: Item_state defaults to the icon_state.

**Comment by Gef21** [comment]
:(

**Comment by Shadowmech88** [comment]
Why?

**Comment by Gef21** [comment]
I couldn't figure out hoods so Kolith did it, he'll probably do a pull request soon™

**Comment by SysadminJeroen** [comment]
I did a bit of work on toggles.dm myself.

bad8fa6dc36b331ce252fa1035e829ec1b6614af
dc787faf1f141117ce564789f70ca5e562851ebd

**Comment by ghost** [comment]
We got it working, and with a couple nicer more intuitive features - eg if you put a hat on, the hood will automatically be taken off, so you don't have to toggle hood to do that. Same with taking the hood off the normal way.

**Comment by Intigracy** [comment]
`>can't wear a hat with the hood up`

smh tbh fam

**Pull request closed without merging**